src/lib/nfs_client/fstab_entry_dialog.rb

```diff
@@ -0,0 +1,403 @@
+      # heading for a directory selection dialog
+      dir = Yast::UI.AskForExistingDirectory(dir, _("Select the Mount Point"))
+
+      return if !dir || dir.empty?
+      Yast::UI.ChangeWidget(Id(:mountent), :Value, dir)
+    end
+
+    def ok_handler
+      server = FormatHostnameForFstab(
+        Yast::UI.QueryWidget(Id(:serverent), :Value)
+      )
+      @pth = StripExtraSlash(
+        Yast::UI.QueryWidget(Id(:pathent), :Value)
+      )
+      @mount = StripExtraSlash(
+        Yast::UI.QueryWidget(Id(:mountent), :Value)
+      )
+      @nfs4 = Yast::UI.QueryWidget(Id(:nfs4), :Value)
+      @nfs41 = Yast::UI.QueryWidget(Id(:nfs41), :Value)
+      @options = Yast::UI.QueryWidget(Id(:optionsent), :Value).tr(" ", "")
+      @options = Yast::NfsOptions.set_nfs41(@options, @nfs41)
+
+      options_error = Yast::NfsOptions.validate(@options)
+      if !CheckHostName(server)
+        Yast::UI.SetFocus(Id(:serverent))
+      elsif !CheckPath(@pth)
+        Yast::UI.SetFocus(Id(:pathent))
+      elsif !CheckPath(@mount) || IsMpInFstab(@existing, @mount)
+        Yast::UI.SetFocus(Id(:mountent))
+      elsif options_error.size > 0
+        Yast::Popup.Error(options_error)
+        Yast::UI.SetFocus(Id(:optionsent))
+      else
+        @fstab_ent = {
+          "spec"    => "#{server}:#{@pth}",
+          "file"    => @mount,
+          "vfstype" => @nfs4 ? "nfs4" : "nfs",
+          "mntops"  => @options
+        }
+        @fstab_ent["old"] = @old unless @old == @fstab_ent["spec"]
+        Yast::Wizard.RestoreScreenShotName
+        log.info "FstabEntryDialog returns #{@fstab_ent}"
+        finish_dialog(@fstab_ent)
+      end
+    end
+
+    def help_handler
+      # help text 1/4
+      # change: locally defined -> servers on LAN
+      helptext = _(
+        "<p>Enter the <b>NFS Server Hostname</b>.  With\n" \
+          "<b>Choose</b>, browse through a list of\n" \
+          "NFS servers on the local network.</p>\n"
+      )
+      # help text 2/4
+      # added "Select" button
+      helptext << _(
+        "<p>In <b>Remote File System</b>,\n" \
+          "enter the path to the directory on the NFS server.  Use\n" \
+          "<b>Select</b> to select one from those exported by the server.\n" \
+          "</p>"
+      )
+      # help text 3/4
+      helptext << _(
+        "<p>\t\t\n" \
+          "For <b>Mount Point</b>, enter the path in the local " \
+          "file system where the directory should be mounted. With\n" \
+          "<b>Browse</b>, select your mount point\n" \
+          "interactively.</p>"
+      )
+      # help text 4/4
+      helptext << _(
+        "<p>For a list of <b>Options</b>,\nread the man page mount(8).</p>"
+      )
+      # popup heading
+      Yast::Popup.LongText(_("Help"), RichText(helptext), 50, 18)
+    end
+
+    protected
+
+    def create_dialog
+      super
+      Yast::UI.ChangeWidget(Id(:serverent), :Value, @server)
+      Yast::UI.SetFocus(Id(:serverent))
```

Comment:
I don't know. Just thinking aloud. Somehow it looked to me not so different from ```dialog_options```. Maybe having a (by default empty) method ```dialog_init``` or something like that to be called at the end ```create_dialog```. Just in case is a common pattern to perform such initialization actions on the widgets. I'm fine with overloading ```create_dialog``` if it's something we are going to do rarely.